plugins/notifications-backend/migrations/20240531_metadata.js

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright 2024 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+exports.up = async function up(knex) {
+  await knex.schema.createTable('notification_metadata', table => {
+    table
+      .uuid('notification_id')
+      .references('id')
+      .inTable('notification')
+      .onDelete('CASCADE')
+      .notNullable()
+      .comment('The entity that provided the relation');
+    table.string('name', 255).notNullable();
+    table.string('type').notNullable();
+    table.text('value').notNullable();
+
+    table.string('name_lower', 255).notNullable();
```

Comment:
...and i guess possibly a separate search table like we have for entities in the catalog, if it's commonly needed to filter among these at the database level

Reply:
Yup, for our company the most important fields are in this table, not in the notification table.

Ah okie, didn't pay attention that this table exists 😃

Reply:
After checking this and thinking a bit about it, it seems that "notification_metadata" already looks like the "search",
instead of having an extra column, yet another table will make the complexity even higher.